routing/result_interpretation.go

```diff
@@ -76,8 +80,9 @@ type interpretedResult struct {
 
 // interpretResult interprets a payment outcome and returns an object that
 // contains information required to update mission control.
-func interpretResult(rt *mcRoute, success bool, failureSrcIdx *int,
-	failure lnwire.FailureMessage) *interpretedResult {
+func interpretResult(rt *mcRoute, success bool,
```

Comment:
hmmm I dont think they are duplicates/redundancies. There are times when we want to report a failure but we don't know the index of the failure. See [this call](https://github.com/lightningnetwork/lnd/blob/acbb33bb7b11e2dd3f2f85fe14fd3ff087212fc0/routing/payment_lifecycle.go#L889) where we are reporting a failure but we set the source index and failure messages to nil

Reply:
right, but in the end we only have successes or failures, so we could combine the failure index and failure message into an optional TLV sub-struct whose presence encodes if it was a failure? but non-blocking, as it's not a huge gain in savings, but it would enforce that we can't have a success and failures set in the same result

Reply:
ah! good point - cool will update 👍